File: app/src/main/java/com/cn29/aac/ui/common/AlertDialogComponent.java
package com.cn29.aac.ui.common;

import android.app.AlertDialog;
import android.content.Context;

import androidx.lifecycle.Lifecycle;
import androidx.lifecycle.LifecycleObserver;
import androidx.lifecycle.OnLifecycleEvent;

/**
 * Created by Charles Ng on 12/10/2017.
 */

public class AlertDialogComponent implements LifecycleObserver {

  private Context context;
  private AlertDialog alertDialog;

  public AlertDialogComponent(Context context, Lifecycle lifecycle) {
    this.context = context;
    lifecycle.addObserver(this);
  }

  public void showDialog(AlertDialog alertDialog) {
    this.alertDialog = alertDialog;
    alertDialog.show();
  }

  public void hideDialog() {
    if (alertDialog != null) {
      alertDialog.dismiss();
    }
  }

  @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
  void onCreate() {
  }

  @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
  void onPause() {
    if (alertDialog != null) {
      alertDialog.dismiss();
    }
  }

  @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
  void stop() {
    if (alertDialog != null) {
      alertDialog.dismiss();
    }

  }
}

File: app/src/main/java/com/cn29/aac/ui/common/AlertDialogComponent.kt
package com.cn29.aac.ui.common

import android.app.AlertDialog
import android.content.Context
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.OnLifecycleEvent

/**
 * Created by Charles Ng on 12/10/2017.
 */
class AlertDialogComponent(private val context: Context,
                           lifecycle: Lifecycle) :
        LifecycleObserver {
    private var alertDialog: AlertDialog? = null
    fun showDialog(alertDialog: AlertDialog) {
        this.alertDialog = alertDialog
        alertDialog.show()
    }

    fun hideDialog() {
        if (alertDialog != null) {
            alertDialog!!.dismiss()
        }
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_CREATE)
    fun onCreate() {
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_PAUSE)
    fun onPause() {
        if (alertDialog != null) {
            alertDialog!!.dismiss()
        }
    }

    @OnLifecycleEvent(Lifecycle.Event.ON_STOP)
    fun stop() {
        if (alertDialog != null) {
            alertDialog!!.dismiss()
        }
    }

    init {
        lifecycle.addObserver(this)
    }
}